### PR TITLE
Add Wait For URL

### PR DESCRIPTION
**Pull request by SeamusMacRat**
Added method for interstitial page loads

### CODE REVIEW

File: dotnet/Liberator.Driver/RatDriverMethods.cs [hunk 1]
```diff
@@ -364,6 +364,61 @@ public void ClickLinkAndWait(By locator, [Optional, DefaultParameterValue(true)]
...
+                {
+                    Element.SendKeys(Keys.Enter);
+                    WaitForUrlToContain(url);
+                }
+                else
+                {
+                    //TODO: IE & Opera currently not reporting staleness. To be investigated.
+                    Element.SendKeys(Keys.Enter);
+                    WaitForUrlToContain(url);
+                }
```
Comment (delahayeoglafa) [inline dotnet/Liberator.Driver/RatDriverMethods.cs]: What is the difference between the if & else statement here? They are both doing the same thing

Comment (SeamusMacRat) [reply dotnet/Liberator.Driver/RatDriverMethods.cs]: Agreed … removing this

File: dotnet/Liberator.Driver/RatDriverMethods.cs [hunk 2]
```diff
@@ -322,16 +322,10 @@ public void ClickLinkAndWait(IWebElement element, [Optional, DefaultParameterVal
             try
```
Comment (delahayeoglafa) [inline dotnet/Liberator.Driver/RatDriverMethods.cs]: Im not sure we need the waitForTarget bool - reasoning: The method name implies that the user has already selected a true for the wait as the method name is ClickLinkAndWait. If the method name was just ClickLink then user might want to control if there is a wait or not - but ClickLinkAndWait implies that there is always going to be a wait. Thoughts?

File: dotnet/Liberator.Driver/RatDriverMethods.cs [hunk 3]
```diff
@@ -364,6 +357,50 @@ public void ClickLinkAndWait(By locator, [Optional, DefaultParameterValue(true)]
             }
         }
 
+        /// <summary>
+        /// Clicks on a link and waits for a new page to be loaded that contains a specified URL or part URL
+        /// </summary>
+        /// <param name="element">The element on which to click</param>
+        /// <param name="url">The partial URL to be waited for</param>
+        /// <param name="waitForTarget">(Optional parameter) Whether to wait for the cliackability of the target element</param>
+        public void ClickLinkAndWaitForUrl(IWebElement element, string url, [Optional, DefaultParameterValue(true)] bool waitForTarget)
```
Comment (delahayeoglafa) [inline dotnet/Liberator.Driver/RatDriverMethods.cs]: Again, we dont need the wait for target bool as we are not using it

File: dotnet/Liberator.Driver/RatDriverMethods.cs [hunk 4]
```diff
@@ -364,6 +357,50 @@ public void ClickLinkAndWait(By locator, [Optional, DefaultParameterValue(true)]
...
+        }
+
+        /// <summary>
+        /// Clicks on a link and waits for a new page to be loaded that contains a specified URL or part URL
+        /// </summary>
+        /// <param name="locator">The locator for the element on which to click</param>
+        /// <param name="url">The partial URL to be waited for</param>
+        /// <param name="wait">(Optional parameter) Whether to wait for the cliackability of the target element</param>
+        public void ClickLinkAndWaitForUrl(By locator, string url, [Optional, DefaultParameterValue(true)] bool wait)
+        {
```
Comment (delahayeoglafa) [inline dotnet/Liberator.Driver/RatDriverMethods.cs]: Same thing as above with the wait boolean

File: dotnet/Liberator.Driver/RatDriverMethods.cs (hunk 2 above)
Comment (SeamusMacRat) [reply dotnet/Liberator.Driver/RatDriverMethods.cs]: Changed

File: dotnet/Liberator.Driver/RatDriverMethods.cs (hunk 3 above)
Comment (SeamusMacRat) [reply dotnet/Liberator.Driver/RatDriverMethods.cs]: Changed

File: dotnet/Liberator.Driver/RatDriverMethods.cs (hunk 4 above)
Comment (SeamusMacRat) [reply dotnet/Liberator.Driver/RatDriverMethods.cs]: Changed

File: dotnet/Liberator.Driver/RatDriverMethods.cs [hunk 5]
```diff
@@ -322,16 +322,10 @@ public void ClickLinkAndWait(IWebElement element, [Optional, DefaultParameterVal
             try
             {
                 LastPage = Driver.FindElement(By.TagName("html"));
-                if(waitForTarget) { WaitForElementToBeClickable(element); }
-                if (typeof(TWebDriver) == typeof(OperaDriver) || typeof(TWebDriver) == typeof(InternetExplorerDriver))
+                if (waitForTarget) { WaitForElementToBeClickable(element); }
+                    ClickLink(element, true);
+                if (typeof(TWebDriver) != typeof(OperaDriver) && typeof(TWebDriver) != typeof(InternetExplorerDriver))
```
[the remark this answers is not in the document]
Comment (SeamusMacRat) [reply dotnet/Liberator.Driver/RatDriverMethods.cs]: Changed

**Pull request merged**